### PR TITLE
Bug node perm check dataclasses

### PR DESCRIPTION
**Pull request by vEpiphyte**
Dataclasses implementation for getAllowedReason that is ever so slightly faster and bundles the string formatting into the dataclass.

### CODE REVIEW

File: synapse/lib/storm.py [hunk 1]
```diff
@@ -2209,30 +2210,32 @@ def confirmPropSet(self, prop, layriden=None):
...
+                if len(meta0.rule) >= len(meta1.rule):
                     return
                 self.user.raisePermDeny(prop.setperms[0], gateiden=layriden)
             return
 
         if allowed1:
             if allowed0 is None:
                 return
-            # allowed0 here is False. This is a deny-with-precedence case.
+            # allowed1 here is False. This is a deny-with-precedence case.
```
Comment (Cisphyx) [inline synapse/lib/storm.py:2236]: isn't allowed0 False here?

Comment (Cisphyx) [inline synapse/lib/storm.py:2236]:
```suggestion
            # allowed0 here is False. This is a deny-with-precedence case.
```

**Review by Cisphyx — APPROVED** [review]

File: synapse/lib/storm.py [hunk 2]
```diff
@@ -2242,30 +2248,32 @@ def confirmPropDel(self, prop, layriden=None):
         if layriden is None:
             layriden = self.snap.wlyr.iden
 
-        allowed0, meta0 = self.allowedReason(prop.delperms[0], gateiden=layriden)
+        meta0 = self.allowedReason(prop.delperms[0], gateiden=layriden)
+        allowed0 = meta0.value
```
Comment (Cisphyx) [inline synapse/lib/storm.py:2252]: can move this one down too

File: synapse/lib/storm.py [hunk 3]
```diff
@@ -1824,6 +1825,8 @@ class Runtime(s_base.Base):
     opaque object which is called through, but not dereferenced.
 
     '''
+
+    _admin_reason = s_hiveauth._allowedReason(True, isadmin=True)
```
Comment (Cisphyx) [inline synapse/lib/storm.py:1829]: could we just put this in s_hiveauth?

Comment (vEpiphyte) [reply synapse/lib/storm.py:1829]: We could but we'll rip it out in the not distant future.

Comment (Cisphyx) [reply synapse/lib/storm.py:1829]: yeah good point

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**